windows-agent/internal/daemon/daemon_test.go

```diff
@@ -144,6 +144,46 @@ func TestStartQuit(t *testing.T) {
 	}
 }
 
+func TestServeNoWSLIP(t *testing.T) {
+	// This test is not parallel because it modifies a global flag that
+	// affects the behavior of the getWslIP function.
+	daemon.SetWslIPErr(t)
+
+	ctx := context.Background()
+	addrDir := t.TempDir()
+
+	registerer := func(context.Context) *grpc.Server {
+		server := grpc.NewServer()
+		var service testGRPCService
+		grpctestservice.RegisterTestServiceServer(server, service)
+		return server
+	}
+
+	d := daemon.New(ctx, registerer, addrDir)
+	defer d.Quit(ctx, false)
+
+	go func() { _ = d.Serve(ctx) }()
```

Comment:
should we have a channel that we close once finishing serving to ensure that we quit properly? (that way, we don’t potentially leak a goroutine even if those are for tests)